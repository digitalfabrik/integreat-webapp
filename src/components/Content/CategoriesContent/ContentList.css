@import url('../../Common/Helper.css');
@import url('../../Common/Colors.css');

.list a {
  @mixin remove-a;
}

.row {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.category-thumbnail {
  width: 50px;
  height: 50px;
  flex-shrink: 0;
  padding: 5px;
}

.category-caption {
  height: 100%;
  min-width: 1px; /* needed to enabled line breaks for to long words, exact value doesn't matter, @Max: DO NOT CHANGE */
  flex-grow: 1;
  margin-left: 15px;
  padding: 15px 0;
  border-bottom: 2px solid var(--theme-color);
  word-wrap: break-word;
}
