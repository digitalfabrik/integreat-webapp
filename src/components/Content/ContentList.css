@import url('../Helper/Helper.css');

.list a {
  @mixin remove-a;
}

.element {
  height: 50px;
}

.element-text {
  height: 100%;
  min-width: 1px; /* exact value doesn't matter, but this lets long words wrap */
  flex-grow: 1;
  margin-left: 15px;
  padding: 15px 0;
  border-bottom: 2px solid #fbda16;
  word-wrap: break-word;
}

.row {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.element-image {
  width: 50px;
  flex-shrink: 0;
  padding: 5px;
}

.image {
  width: 50px;
  max-width: 100%;
  max-height: 100%;
}
