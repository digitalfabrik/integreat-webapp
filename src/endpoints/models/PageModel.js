class PageModel {
  constructor ({id, numericId, title = '', parentNumericId = null, content = '', thumbnail = null, order = 0, availableLanguages = {}, url = null}) {
    this._id = id
    this._numericId = numericId
    this._title = title
    this._content = content
    this._parentNumericId = parentNumericId
    this._thumbnail = thumbnail
    this._order = order
    this._availableLanguages = availableLanguages
    this._children = []
    this._parent = null
    this._url = url
  }

  get thumbnail () {
    return this._thumbnail
  }

  addChild (page) {
    this._children.push(page)
    this._children.sort((page1, page2) => page1.order - page2.order)
  }

  setParent (parent) {
    this._parent = parent
  }

  get parent () {
    return this._parent
  }

  get id () {
    return this._id
  }

  get numericId () {
    return this._numericId
  }

  get title () {
    return this._title
  }

  get content () {
    return this._content
  }

  get parentNumericId () {
    return this._parentNumericId
  }

  get children () {
    return this._children
  }

  get order () {
    return this._order
  }

  get availableLanguages () {
    return this._availableLanguages
  }

  get url () {
    if (this._url) {
      return this._url
    }
    return `${this.parent.url}/${this.id}`
  }
}

export const EMPTY_PAGE = new PageModel({})
export default PageModel
