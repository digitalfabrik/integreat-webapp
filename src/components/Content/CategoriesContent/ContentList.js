import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'redux-little-router'

import style from './ContentList.css'
import PageModel from 'endpoints/models/PageModel'

import IconPlaceholder from './assets/IconPlaceholder.svg'

class SecondOrderCategory extends React.Component {
  static propTypes = {
    page: PropTypes.instanceOf(PageModel).isRequired,
    url: PropTypes.string.isRequired
  }

  render () {
    return (
      <Link href={this.props.url}>
        <div className={style.row}>
          <img className={style.categoryThumbnail} src={this.props.page.thumbnail || IconPlaceholder}/>
          <div className={style.categoryCaption}>{this.props.page.title}</div>
        </div>
      </Link>
    )
  }
}

class ContentList extends React.Component {
  static propTypes = {
    pages: PropTypes.arrayOf(PropTypes.shape({
      page: PropTypes.instanceOf(PageModel).isRequired,
      url: PropTypes.string.isRequired
    })).isRequired
  }

  render () {
    return (
      <div className={style.list}>
        { this.props.pages.map(({ page, url }) => <SecondOrderCategory key={url} url={url} page={page} />) }
      </div>
    )
  }
}

export default ContentList
