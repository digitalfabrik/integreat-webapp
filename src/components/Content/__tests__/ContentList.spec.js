import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import PageModel from '../../../endpoints/models/PageModel'
import ContentList from '../ContentList'
import store from '../../../store'

describe('ContentList', () => {
  const page4 = new PageModel({
    id: 'page4',
    numericId: 4,
    title: 'Page4',
    parent: 2
  })

  const page2 = new PageModel({
    id: 'page2',
    numericId: 2,
    title: 'Page2',
    parent: 0,
    children: [page4]
  })

  const page3 = new PageModel({
    id: 'page3',
    numericId: 3,
    title: 'Page3',
    parent: 0
  })

  const pages = [
    {page: page2, url: '/augsburg/de/page2'},
    {page: page3, url: '/augsburg/de/page3'},
    {page: page4, url: '/augsburg/de/page3/page4'}
  ]

  test('should render', () => {
    const wrapper = renderer.create(
      <Provider store={store}>
        <ContentList pages={pages} />
      </Provider>)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
