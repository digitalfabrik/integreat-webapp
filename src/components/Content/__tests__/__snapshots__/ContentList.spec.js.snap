// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentList should render 1`] = `
<div
  className="list"
>
  <a
    href="/augsburg/de/page2"
    onClick={[Function]}
    target={undefined}
  >
    <div
      className="row"
    >
      <div
        className="elementImage element"
      >
        <img
          className="image"
          src={null}
        />
      </div>
      <div
        className="elementText"
      >
        Page2
      </div>
    </div>
  </a>
  <a
    href="/augsburg/de/page3"
    onClick={[Function]}
    target={undefined}
  >
    <div
      className="row"
    >
      <div
        className="elementImage element"
      >
        <img
          className="image"
          src={null}
        />
      </div>
      <div
        className="elementText"
      >
        Page3
      </div>
    </div>
  </a>
  <a
    href="/augsburg/de/page3/page4"
    onClick={[Function]}
    target={undefined}
  >
    <div
      className="row"
    >
      <div
        className="elementImage element"
      >
        <img
          className="image"
          src={null}
        />
      </div>
      <div
        className="elementText"
      >
        Page4
      </div>
    </div>
  </a>
</div>
`;
